tests/components/balboa/test_config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Test the Balboa Spa config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.balboa.config_flow import AlreadyConfigured, CannotConnect
+from homeassistant.components.balboa.const import DOMAIN
+
+from tests.common import mock_coro
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.balboa.config_flow.validate_input",
```

Comment:
Please patch the external code instead.
`homeassistant.components.balboa.config_flow.BalboaSpaWifi`